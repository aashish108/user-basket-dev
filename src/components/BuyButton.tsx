import styled from '@emotion/styled';
import { useContext, useEffect, useState } from 'react';
import { Context } from '../contexts/Context';

const ButtonContainer = styled.div`
 display: flex;
 flex-direction: row;
 justify-content: end;
`;
const Button = styled.button`
 background-color: green;
 color: white;
 padding: 10px;
 margin-top: 20px;
 width: 100px;
 &:hover {
  background-color: darkgreen;
 }
`;

// type A = { name: string };

const BuyButton = () => {
 const { basketItems } = useContext(Context);
 const [buyButtonDisabled, setBuyButtonDisabled] = useState(false);

 const buy = () => {
  console.log('Buying!');
 };

 useEffect(() => {
  if (basketItems.length === 0) {
   return setBuyButtonDisabled(true);
  }
  return setBuyButtonDisabled(false);
 }, [basketItems]);

 return (
  <ButtonContainer>
   <Button onClick={buy} disabled={buyButtonDisabled}>
    Buy
   </Button>
  </ButtonContainer>
 );
};

export default BuyButton;
