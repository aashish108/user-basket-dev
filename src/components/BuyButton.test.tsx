import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react';
import BuyButton from './BuyButton';

afterEach(cleanup);

it('renders correctly', async () => {
 const component = renderer.create(<BuyButton />);
 let tree = component.toJSON();
 expect(tree).toMatchSnapshot();

 render(<BuyButton />);
 const text = await screen.findAllByText(/Buy/);
 expect(text).toBeTruthy();
});
