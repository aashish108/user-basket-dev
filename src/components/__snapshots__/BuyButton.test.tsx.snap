// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1mtu2dh"
>
  <button
    className="css-lvrehb"
    disabled={false}
    onClick={[Function]}
  >
    Buy
  </button>
</div>
`;

exports[`shows the correct button text 1`] = `
<div
  className="css-1mtu2dh"
>
  <button
    className="css-lvrehb"
    disabled={false}
    onClick={[Function]}
  >
    Buy
  </button>
</div>
`;
